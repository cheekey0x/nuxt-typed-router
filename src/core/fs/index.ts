export * from './prettierFormat';
export * from './writeFile';
