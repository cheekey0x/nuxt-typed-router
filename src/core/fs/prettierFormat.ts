import chalk from 'chalk';
import logSymbols from 'log-symbols';

const defaultPrettierOptions = {
  printWidth: 100,
  tabWidth: 2,
  trailingComma: 'es5',
  singleQuote: true,
  semi: true,
  bracketSpacing: true,
  htmlWhitespaceSensitivity: 'strict',
} as const;

// export function formatOutputWithPrettier(template: string): string {
// try {
//   let prettierFoundOptions = await resolveConfig(process.cwd());

//   if (!prettierFoundOptions) {
//     prettierFoundOptions = defaultPrettierOptions;
//   }

//   const formatedTemplate = format(template, {
//     ...prettierFoundOptions,
//     parser: 'typescript',
//   });

//   return template;
// } catch (e) {
//   console.error(logSymbols.error, chalk.red('Error while formatting the output'), '\n' + e);
//   return Promise.reject(e);
// }
// }
