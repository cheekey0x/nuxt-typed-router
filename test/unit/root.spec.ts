import { readFile } from 'fs/promises';
import { createPage, setupTest } from '@nuxt/test-utils';
import 'core-js';

describe('Init route file', () => {
  // setupTest({
  //   browser: true,
  //   build: true,
  //   setupTimeout: 100000,
  //   rootDir: 'test/fixtures/basic',
  //   config: {
  //     server: {
  //       port: 2310,
  //     },
  //   },
  // });
  it('should generate to correct tree for the pages folder', async () => {
    const routeFile = await readFile('playground/generated/__routes.ts', 'utf-8');
    const definitionsFile = await readFile('playground/generated/typed-router.d.ts', 'utf-8');
    expect(routeFile).toMatchSnapshot();
    expect(definitionsFile).toMatchSnapshot();
  });
});
