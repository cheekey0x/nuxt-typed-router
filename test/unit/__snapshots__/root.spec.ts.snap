// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Init route file should generate to correct tree for the pages folder 1`] = `
"export const routerPagesNames = {
  activate: 'activate',
  rootPage: 'rootPage',
  childOne: {
    index: 'parent-child-one',
    childOneChildOneSubOne: 'parent-child-one-child-one-sub-one',
    childOneChildOneSubTwo: 'parent-child-one-child-one-sub-two',
    user: { index: 'parent-child-one-child-one-sub-one-user' },
  },
  childTwo: {
    index: 'parent-child-two',
    childTwoChildOneSubOne: 'parent-child-two-child-one-sub-one',
  },
  index: 'index',
  all: 'all',
};
"
`;
